pkg/controller/pod.go

```diff
@@ -202,15 +214,87 @@ func (c *Controller) syncPod(ctx context.Context, pod *corev1.Pod) error {
 		}
 	}
 
-	klog.V(4).Infof("Registering instance %s under service %+v", pod.Name, cloudmapConfig)
+	var serviceItem *CloudMapInstanceCacheItem
+	serviceInstanceSummary := make(map[string]bool)
+	cloudMapServiceKey := *cloudmapConfig.NamespaceName + "-" + *cloudmapConfig.ServiceName
```

Comment:
it's better to check non-nil for cloudmapConfig.NamespaceName & cloudmapConfig.ServiceName to protect us from nil values. (seems it's not checked other where, correct me if i'm wrong)
otherwise a VN without specified NamespaceName will crash the controller.